src/services/debug-service.ts

```diff
@@ -0,0 +1,177 @@
+        if (error) {
+          console.error(consoleMessage, { details, error });
+        } else {
+          console.error(consoleMessage, details);
+        }
+        break;
+      case 'warn':
+        console.warn(consoleMessage, details);
+        break;
+      case 'info':
+      default:
+        console.log(consoleMessage, details);
+        break;
+    }
+  }
+
+  static getLogs(): DebugLogEntry[] {
+    return [...this.logs].reverse(); // Return newest first
+  }
+
+  static clearLogs(): void {
+    this.logs = [];
+  }
+
+  static async getAppState(): Promise<DebugAppState> {
+    try {
+      const bookmarks = await DatabaseService.getAllBookmarks();
+      const assets = await DatabaseService.getAllAssets();
+      const settings = await DatabaseService.getSettings();
+      const lastSyncTimestamp = await DatabaseService.getLastSyncTimestamp();
+
+      // Calculate storage usage if supported
+      let storageInfo = {};
+      if ('storage' in navigator && 'estimate' in navigator.storage) {
+        try {
+          const estimate = await navigator.storage.estimate();
+          storageInfo = {
+            sizeEstimate: estimate.usage,
+            quotaUsed: estimate.usage,
+            quotaAvailable: estimate.quota
+          };
+        } catch (error) {
+          // Storage estimate not available
+        }
+      }
+
+      // Get bookmarks with assets
+      const bookmarksWithAssets = new Set(assets.map((a: any) => a.bookmark_id));
+
+      return {
+        bookmarks: {
+          total: bookmarks.length,
+          unread: bookmarks.filter(b => b.unread).length,
+          archived: bookmarks.filter(b => b.is_archived).length,
+          withAssets: bookmarksWithAssets.size
+        },
+        sync: {
+          isInProgress: false, // Will be updated by sync service
+          ...(lastSyncTimestamp && { lastSyncAt: lastSyncTimestamp })
+        },
+        api: {
+          ...(settings?.linkding_url && { baseUrl: settings.linkding_url })
+        },
+        storage: storageInfo
+      };
+    } catch (error) {
+      this.log('error', 'app', 'getAppState', 'Failed to get app state', undefined, error instanceof Error ? error : new Error(String(error)));
+      throw error;
+    }
+  }
+
+  // Convenience methods for common logging scenarios
+  static logSyncStart(total: number): void {
+    this.log('info', 'sync', 'start', `Starting sync of ${total} bookmarks`, { total });
+  }
+
+  static logSyncProgress(current: number, total: number): void {
+    this.log('info', 'sync', 'progress', `Sync progress: ${current}/${total}`, { current, total });
+  }
+
+  static logSyncComplete(processed: number): void {
+    this.log('info', 'sync', 'complete', `Sync completed successfully`, { processed });
+  }
+
+  static logSyncError(error: Error, context?: any): void {
+    this.log('error', 'sync', 'error', 'Sync failed', context, error);
+  }
+
+  static logApiRequest(url: string, method: string): void {
```

Comment:
It looks like only the sync service was updated to use these new logging methods. Update the rest of the app (API requests, database errors, etc) as well.